src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java

```diff
@@ -5,70 +5,59 @@
 import com.aws.greengrass.telemetry.models.TelemetryUnit;
 
-import java.time.Instant;
+import java.time.Clock;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicLong;
+import javax.inject.Inject;
 
 public class ClientDeviceAuthMetrics {
+    private final AtomicLong subscribeToCertificateUpdatesSuccess = new AtomicLong();
+    private final MetricFactory mf = new MetricFactory(NAMESPACE);
+    private final Clock clock;
+    private static final String NAMESPACE = "ClientDeviceAuth";
 
-    private final AtomicLong certSubscribeSuccess = new AtomicLong();
-    private final AtomicLong certSubscribeError = new AtomicLong();
-    private final AtomicLong invalidConfig = new AtomicLong();
-    private final AtomicLong certRotation = new AtomicLong();
+    /**
+     * Constructor for Client Device Auth Metrics.
+     *
+     * @param clock Clock
+     */
+    @Inject
+    public ClientDeviceAuthMetrics(Clock clock) {
+        this.clock = clock;
+    }
 
-    private static final String NAMESPACE = "ClientDeviceAuth";
+    /**
+     * Emit metrics using Metric Factory.
+     */
+    public void emitMetrics() {
+        List<Metric> retrievedMetrics = collectMetrics();
+        for (Metric retrievedMetric : retrievedMetrics) {
+            mf.putMetricData(retrievedMetric);
+        }
+    }
 
     /**
      * Builds the CDA metrics.
+     *
      * @return a list of {@link Metric}
      */
     public List<Metric> collectMetrics() {
         List<Metric> metricsList = new ArrayList<>();
 
-        long timestamp = Instant.now().toEpochMilli();
+        long timestamp = clock.instant().toEpochMilli();
 
         Metric metric = Metric.builder()
                 .namespace(NAMESPACE)
-                .name("Cert.SubscribeSuccess")
-                .unit(TelemetryUnit.Count)
-                .aggregation(TelemetryAggregation.Sum)
-                .value(certSubscribeSuccess.getAndSet(0))
-                .timestamp(timestamp)
-                .build();
-        metricsList.add(metric);
-
-        metric = Metric.builder()
-                .namespace(NAMESPACE)
-                .name("Cert.SubscribeError")
-                .unit(TelemetryUnit.Count)
-                .aggregation(TelemetryAggregation.Sum)
-                .value(certSubscribeError.getAndSet(0))
-                .timestamp(timestamp)
-                .build();
-        metricsList.add(metric);
-
-        metric = Metric.builder()
-                .namespace(NAMESPACE)
-                .name("Config.Invalid")
-                .unit(TelemetryUnit.Count)
-                .aggregation(TelemetryAggregation.Sum)
-                .value(invalidConfig.getAndSet(0))
-                .timestamp(timestamp)
-                .build();
-        metricsList.add(metric);
-
-        metric = Metric.builder()
-                .namespace(NAMESPACE)
-                .name("Cert.Rotate")
+                .name("SubscribeToCertificateUpdates.Success")
```

Comment:
since this is duplicated, let's get a public constant for it.

Reply:
I would need to inject a `ClientDeviceAuth` instance into `CertificateManager` to use the public constant. Would doing that be ok even though we're decoupling them?

Reply:
Ideally this _isn't_ duplicated. Only the the thing sending this to the telemetry emitter needs to know it